test/main_contracts/rate_oracles/compoundBorrowRateOracle.ts

```diff
@@ -0,0 +1,129 @@
+import { ethers, waffle } from "hardhat";
```

Comment:
@CyclopsRex @0xZenus this needs a very careful review

Reply:
It seems OK but:

1. it might be worth moving the logic to calculate the expected rate into its own function for easy re-use
 
2. I would feel more comfortable if we were also doing some testing from first principles. E.g.

 - choose an APY, `a`
 - set the block rate as we think is required to effect that APY
 - write some rates with various block deltas
 - check that the APY we calculate pairwise between any of the chosen written data points is equal to `a`, perhaps with some small allowance for compounding.

Reply:
I implemented this test using a series of different block deltas and rates